integration/src/main/java/com/aws/greengrass/mqtt/moquette/ClientDeviceAuthorizer.java

```diff
@@ -201,4 +224,37 @@ private void closeAuthSession(String clientId, String username) {
             }
         }
     }
+
+    /**
+     * Emits MQTT AuthError metrics for requested MQTT operation.
+     * Ideally these metrics should be emitted from the Moquette request handlers.
+     * Emitting metrics from Authorizer integration to avoid broker code change.
+     *
+     * @param operation Requested MQTT operation
+     */
+    private void emitAuthErrorMetric(String operation) {
+        String authErrorMetric;
+        switch (operation) {
+            case "mqtt:connect":
+                authErrorMetric = MqttMetrics.CONNECT_AUTH_ERROR;
+                break;
+            case "mqtt:publish" :
+                authErrorMetric = MqttMetrics.PUBLISH_IN_AUTH_ERROR;
+                break;
+            case "mqtt:subscribe" :
+                authErrorMetric = MqttMetrics.SUBSCRIBE_AUTH_ERROR;
+                break;
+            default:
+                authErrorMetric = MqttMetrics.UNKNOWN_AUTH_ERROR;
+                break;
+        }
+        metricsEmitter.emitMetric(Metric.builder()
```

Comment:
we probably do not want to immediately emit a metric, this can be expensive. We may want to instead batch up a number of them and then emit the sum ourselves on a timer.

Reply:
updated to schedule batched metric emit